### PR TITLE
Code of Conduct

### PR DESCRIPTION
**Pull request by Oliver-makes-code**
This is the code of conduct for Helix, it is an initial draft. Discussion on what we should add, remove, or changed should be within this pull request

### CODE REVIEW

**Review by Ecorous — APPROVED** [review]
Looks good, if this is about server rules as well remove piracy stuff

**Comment by Ecorous** [comment]
*ban piracy stuff

**Comment by intergrav** [comment]
Looks fine to me

**Review by Astralchroma — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: code-of-conduct.md
Comment (korewaChino) [inline code-of-conduct.md]: inspiration*

Comment (korewaChino) [inline code-of-conduct.md]: inspiration*

**Review by intergrav — APPROVED** [review]
Piracy should be addressed

**Review by korewaChino — CHANGES_REQUESTED** [review]

**Review by korewaChino — APPROVED** [review]

**Review by intergrav — CHANGES_REQUESTED** [review]
Piracy should be addressed somewhere in here, to not discuss/encourage it

**Review by violet-sh — COMMENTED** [review]
I agree with Exo and Devin on piracy stuff, other than that LGTM.

**Comment by Ecorous** [comment]
Sorry if I requested you twice, was just trying to get everyone

**Review by ghost — APPROVED** [review]
Looks good! I do agree with the piracy stuff would be good to have here.

**Comment by Ecorous** [comment]
If I requested you to review again, you don't have to review again, but if you want to you can. We only want at least one review from each requested member

**Review by leo60228 — CHANGES_REQUESTED** [review]
"Being respectful of differing viewpoints, experiences, and opinions" is poorly phrased, and should be rewritten to clearly exclude respecting bigotry.

I disagree with the suggestions from others that the Code of Conduct should explicitly include complying with (a subset of) laws. That is a pragmatic policy decision which should be part of server rules, not a baseline requirement of the community.

**Comment by kadirac** [comment]
https://github.com/HelixLauncher/Community-Guidelines/pull/2 is my draft; will resolve conflicts as needed. i second the things leo said.

**Review by intergrav — APPROVED** [review]

**Comment by cplir-c** [comment]
Is forbidding mention of piracy a good idea? I thought a significant part of the goal of a compliant minecraft launcher is authentication and requiring purchase of the game.
Also, does forbidding discussion of substance use mean trans community members can't talk about hormones or anything?

**Review by kadirac — CHANGES_REQUESTED** [review]
idk how to edit the stuff i said before to a request for change, sorry

**Comment by violet-sh** [comment]
I think regardless of the legalese included in the Code of Conduct (none, all, whatever), there should at least be mention of the fact that there is a EULA agreed upon when downloading and playing the game. Piracy is a strict breaking of this agreement which is a legal offense whether you want to include a requirement to comply with such laws or not. It is, in my opinion at least, a good idea to convey to the user that we as a development team will not be held liable for such things.

I guess there is the argument that this doesn't strictly fall into a "Code of Conduct", but regardless, discussion of things like piracy are things we probably don't want to include ourselves in.

**Comment by Oliver-makes-code** [comment]
> Also, does forbidding discussion of substance use mean trans community members can't talk about hormones or anything?

I could probably specify that it means illegal substances

**Review by Ecorous — APPROVED** [review]

**Review by Astralchroma — APPROVED** [review]

**Review by violet-sh — APPROVED** [review]

**Comment by alikindsys** [comment]
My personal views on piracy land on a gray area, I don't think forbidding the discussion of piracy to be a necessary thing, since as @cplir-c said, to be a compliant launcher you'd need to both authenticate and own the game before launching it. That just feels redundant to me.

Imo, if its decided that offline accounts should be supported (mostly for the benefit of playing the game without an internet connection), we should take the same route the PrismLauncher does, allowing the creation of offline accounts/launching the game offline if you have added a valid microsoft account that owns the game.

Other then that, I vouch for something closer to #2. Unfortunately we need to make it explicit that some "ideas" are inherently harmful and should not be respected.

**Comment by intergrav** [comment]
> Other then that, I vouch for something closer to #2. Unfortunately we need to make it explicit that some "ideas" are inherently harmful and should not be respected.

+1 on this actually, I prefer #2 aswell

**Comment by kadirac** [comment]
alright lemme try to fix conflicts between here and #2 again

**Comment by kadirac** [comment]
Screen Shot 2022-10-22 at 22 23 44
this phrase is so vague i cannot even meaningfully object to it

**Comment by Oliver-makes-code** [comment]
github momen
not letting me rerequest reviews properly

**Review by Ecorous — APPROVED** [review]

**Review by Astralchroma — APPROVED** [review]

**Comment by Fnige** [comment]
Looks pretty good to us

**Review by Fnige — APPROVED** [review]

**Review by alikindsys — APPROVED** [review]

**Review by violet-sh — APPROVED** [review]
LGTM, tho it should probably be un-draftified before merging?

**Review by Ember-Ember — APPROVED** [review]
um, seems good?

**Review by korewaChino — APPROVED** [review]
meh, lgtm

**Review by intergrav — APPROVED** [review]
Looks good!

**Review by Ecorous — APPROVED** [review]

**Review by it0946 — APPROVED** [review]

**Review by kadirac — APPROVED** [review]
it's not complete but i think it's minimally viable. will ping all remaining folk listed in a second; would like to ask that they respond within 48h

**Comment by kadirac** [comment]
> "Being respectful of differing viewpoints, experiences, and opinions" is poorly phrased, and should be rewritten to clearly exclude respecting bigotry.
> 
> I disagree with the suggestions from others that the Code of Conduct should explicitly include complying with (a subset of) laws. That is a pragmatic policy decision which should be part of server rules, not a baseline requirement of the community.

is this resolved to your satisfaction

**Comment by kadirac** [comment]
@ENDERZOMBI102 @j0lol @kb-1000 @pluiedev pardon the ping; see above for context

**Comment by Ecorous** [comment]
> Piracy should be addressed somewhere in here, to not discuss/encourage it

Can you mark your thing as resolved somehow

**Review by ENDERZOMBI102 — APPROVED** [review]
sounds pretty nice

File: code-of-conduct.md
Comment (kb-1000) [inline code-of-conduct.md]: I don't think this sort of wording is appropriate, especially in a CoC.

File: code-of-conduct.md
Comment (kb-1000) [inline code-of-conduct.md]: I'm not sure this list is a good idea, at least in this form. It's not exhaustive, but your wording makes it sound exhaustive.
I'd more suggest something along the lines of (but not word-by-word) "We do not accept bigotry in our spaces", perhaps with "including but not limited to" if you do think that list is necessary.

**Review by kb-1000 — CHANGES_REQUESTED** [review]
some comments on the current draft, but tbh I think the previous draft read nicer and sounded more "professional" - which _is_ a concern for a larger community-based project.
It may a better idea to copy a CoC and make adjustments if necessary rather than rewriting it...

**Review by intergrav — APPROVED** [review]

**Review by FractalGeodes — APPROVED** [review]
Looks good to us

File: code-of-conduct.md
Comment (leo60228) [inline code-of-conduct.md]: I maintain that including "illegal activities" in a Code of Conduct dilutes its value. This should be in a separate document for rules of official community spaces.
Either way, a jurisdiction should be mentioned. Complying with the laws of *every* jurisdiction would necessitate directly breaking other parts of the CoC.

**Review by leo60228 — CHANGES_REQUESTED** [review]

**Review by pluiedev — APPROVED** [review]

**Comment by leo60228** [comment]
> I could probably specify that it means illegal substances

Testosterone is a controlled substance in many countries.

**Review by j0lol — APPROVED** [review]

File: code-of-conduct.md
Comment (kb-1000) [inline code-of-conduct.md]: would read a bit better in plain text if you aligned it:
```suggestion
  opinions, unless they, directly or indirectly, harm a group of
  people
- Educating people about why certain actions are harmful
- Apologizing for, accepting responsibility for, and leaning from
  mistakes
- Focusing on what is best not just for us as individuals, but for
  the overall community
```

File: code-of-conduct.md
Comment (kb-1000) [inline code-of-conduct.md]:
```suggestion
This document is a modified version of the
[QuiltMC Code of Conduct](https://quiltmc.org/en/community/code-of-conduct/),
which itself takes inspiration from the
[Contributor Covenant version 2](https://www.contributor-covenant.org/version/2/0/code_of_conduct.html),
[JetBrains Open Source and Community Code of Conduct](https://confluence.jetbrains.com/display/ALL/JetBrains+Open+Source+and+Community+Code+of+Conduct),
[Microsoft Open Source Code of Conduct](https://microsoft.github.io/codeofconduct),
and [Open Source Design Code of Conduct](https://opensourcedesign.net/code-of-conduct).
```
It would also be possible to convert this to reference style links, moving the actual URLs out of line: 
```suggestion
This document is a modified version of the [QuiltMC Code of Conduct][quiltmc],
which itself takes inspiration from the [Contributor Covenant version 2][cco],
[JetBrains Open Source and Community Code of Conduct][jetbrains],
[Microsoft Open Source Code of Conduct][microsoft],
and [Open Source Design Code of Conduct][opensourcedesign].

[quiltmc]: https://quiltmc.org/en/community/code-of-conduct/
[cco]: https://www.contributor-covenant.org/version/2/0/code_of_conduct.html
[jetbrains]: https://confluence.jetbrains.com/display/ALL/JetBrains+Open+Source+and+Community+Code+of+Conduct
[microsoft]: https://microsoft.github.io/codeofconduct
[opensourcedesign]: https://opensourcedesign.net/code-of-conduct
```

**Review by kb-1000 — COMMENTED** [review]
I also agree with @leo60228, otherwise LGTM.

**Review by FractalGeodes — APPROVED** [review]

File: code-of-conduct.md
Comment (leo60228) [inline code-of-conduct.md]: I think this just reads weirdly at this point.

**Review by leo60228 — APPROVED** [review]
I'd be fine with this iteration, though.

**Review by kb-1000 — APPROVED** [review]

**Comment by kb-1000** [comment]
Since everyone approved and all comments are addressed, I'm going to merge this now.

**Pull request merged**
